primitives/src/pcs/univariate_kzg/mod.rs

```diff
@@ -147,12 +147,18 @@ impl<E: Pairing> PolynomialCommitmentScheme for UnivariateKzgPCS<E> {
         polynomial: &Self::Polynomial,
         point: &Self::Point,
     ) -> Result<(Self::Proof, Self::Evaluation), PCSError> {
+        #[cfg(feature = "kzg-print-trace")]
         let open_time =
             start_timer!(|| format!("Opening polynomial of degree {}", polynomial.degree()));
+
         let divisor = Self::Polynomial::from_coefficients_vec(vec![-*point, E::ScalarField::one()]);
 
+        #[cfg(feature = "kzg-print-trace")]
         let witness_time = start_timer!(|| "Computing witness polynomial");
+
         let witness_polynomial = polynomial / &divisor;
+
+        #[cfg(feature = "kzg-print-trace")]
```

Comment:
note, we don't need to conditioned this under the feature flag every time.

`ark_std::start_timer` has an trivial/empty implementation for `not(features("print-trace"))` as well ([see here](https://github.com/arkworks-rs/std/blob/master/src/perf_trace.rs#L180-L185))

Reply:
Yes but downstream needs `print-trace` for its own timer. This timer has only "log level". I added this feature gate because my downstream logs were flooded with low-level spam. The proper solution is to add some kind of scoping to `start_timer` upstream, but that's beyond our control. 😕

Reply:
I see what you mean. I do remember seeing annoyingly detailed log level when benchmarking plonk gen as well.

Fair enough!